ext/gd/gd.c

```diff
@@ -3939,6 +3949,12 @@ PHP_FUNCTION(imagescale)
 		Z_PARAM_LONG(tmp_h)
 		Z_PARAM_LONG(tmp_m)
 	ZEND_PARSE_PARAMETERS_END();
+
+	if (tmp_m < GD_DEFAULT || tmp_m >= GD_METHOD_COUNT) {
+		zend_argument_value_error(4, "must be a valid mode");
+		RETURN_THROWS();
+	}
```

Comment:
Normally we enumerate what the modes are, or indicate `must be one of GD_*` to make the error message more useful.